File: app/src/main/java/com/byoutline/kickmaterial/dagger/AppComponent.java
package com.byoutline.kickmaterial.dagger;


import com.byoutline.kickmaterial.managers.AccessTokenProvider;
import com.byoutline.secretsauce.di.AppComponentInterface;
import dagger.Component;

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 27.03.15.
 */
@AppScope
@Component(modules = AppModule.class)
public interface AppComponent extends AppComponentInterface {
    AccessTokenProvider getAccessTokenProvider();
}

File: app/src/main/java/com/byoutline/kickmaterial/dagger/AppComponent.kt
package com.byoutline.kickmaterial.dagger

import com.byoutline.kickmaterial.KickMaterialApp
import com.byoutline.kickmaterial.model.DiscoverQuery
import com.byoutline.kickmaterial.model.DiscoverResponse
import com.byoutline.observablecachedfield.ObservableCachedFieldWithArg
import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 27.03.15.
 */
@Singleton
@Component(modules = [
    AppModule::class,
    MainActivityModule::class,
    AndroidSupportInjectionModule::class,
    ViewModelMapModule::class,
    ViewModelProvidersModule::class
])
interface AppComponent {
    fun inject(app: KickMaterialApp)

    val app: KickMaterialApp
    val discoverField: ObservableCachedFieldWithArg<DiscoverResponse, DiscoverQuery>
    @AnimationDurationMultiplier fun getAnimationDurationMultiplier(): Int
}
